vertical-pod-autoscaler/pkg/recommender/routines/cpu_integer_post_processor.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package routines
+
+import (
+	apiv1 "k8s.io/api/core/v1"
+	vpa_types "k8s.io/autoscaler/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1"
+	"k8s.io/autoscaler/vertical-pod-autoscaler/pkg/recommender/model"
+	"math"
+	"strings"
+)
+
+// IntegerCPUPostProcessor ensures that the recommendation delivers an integer value for CPU
+// This is need for users who want to use CPU Management with static policy: https://kubernetes.io/docs/tasks/administer-cluster/cpu-management-policies/#static-policy
+type IntegerCPUPostProcessor struct{}
+
+const (
+	// The user interface for that post processor is an annotation on the VPA object with the following format:
+	// vpa-post-processor.kubernetes.io/{containerName}_integerCPU=true
+	vpaPostProcessorPrefix           = "vpa-post-processor.kubernetes.io/"
+	vpaPostProcessorIntegerCPUSuffix = "_integerCPU"
+	vpaPostProcessorIntegerCPUValue  = "true"
+)
+
+var _ RecommendationPostProcessor = &IntegerCPUPostProcessor{}
```

Comment:
It is a common pattern in Go that express the intention of the developper: the struct `IntegerCPUPostProcessor` must implement `RecommendationPostProcessor` interface.

It does not do anything at runtime, but it is really useful for understanding. Here we have interface and struct in the same program, so that might be obvious, but this turn out to be a very good habit when you create structs that implements should interface of another package (imported from other project). With a good IDE integration with Go you immediately see that you are breaking the contract.

here is an example in the kubernetes repo:
https://github.com/kubernetes/client-go/blob/1ac8d459351e21458fd1041f41e43403eadcbdba/tools/cache/store.go#L147

You can find many other examples with `git grep "var _ .* = &"` in the codebase of kube.